internal/controller/controllers/clusterdeployments_controller.go

```diff
@@ -418,6 +441,30 @@ func (r *ClusterDeploymentsReconciler) createNewCluster(
 	return r.updateState(ctx, clusterDeployment, c, err)
 }
 
+func (r *ClusterDeploymentsReconciler) createNewDay2Cluster(
```

Comment:
when do you set kubeconfig etc?

Reply:
No need to... I made sure with @yevgeny-shnaidman  that it is not needed.
On cloud env we don't have these credentials.

Reply:
So what is the flow of adding workers with CRDs?

Reply:
Should be the same.
Boot from ISO, host register -> Agent CRD created
For the install, there is a different API: InstallHosts that will install all hosts that are in Known state

Reply:
If every host can be installed automatically then maybe it should be done in the host controller? 
Suggesting to talk about it

Reply:
The current API will install all the hosts in the clusters in Known state. It has no granularity:
https://github.com/openshift/assisted-service/blob/master/internal/bminventory/inventory.go#L1277

We can add a new API if needed.

Reply:
We can probably work with the same logic we have right now, if the host is approved we call for install
Not sure what is the side affect if we call it multiple times